onnxruntime/core/providers/cuda/tensor/slice.cc

```diff
@@ -26,9 +35,16 @@ Status Slice::ComputeInternal(OpKernelContext* ctx) const {
   std::vector<int64_t> starts(dimension_count, 0);
   std::vector<int64_t> output_dims(input_dimensions);
 
-  ORT_RETURN_IF_ERROR(PrepareForCompute(attr_starts_, attr_ends_, attr_axes_, 
-			                dimension_count, input_dimensions,
-					starts, output_dims));
+  if (dynamic) {
+    std::vector<int64_t> input_starts, input_ends, input_axes;
+    FillVectorsFromInput<Tind>(ctx, input_starts, input_ends, input_axes);
```

Comment:
>FillVectorsFromInput<Tind>(ctx, input_starts, input_ends, input_axes); [](start = 4, length = 70)

if the data like input_starts and input_ends are read from input, they are in GPU memory, need to write a cuda kernel to get the data.

Reply:
Isn't the copy to CUDA handled below? Seems the logic is shared with static slice

---
In reply to: [246170274](https://github.com/Microsoft/onnxruntime/pull/286#discussion_r246170274) [](ancestors = 246170274)

Reply:
FillVectorsFromInput creates pointer array, and then copied to GPU sharing the logic with static slice. As long as those pointers not dereferenced, the code should be fine.

---
In reply to: [246170274](https://github.com/Microsoft/onnxruntime/pull/286#discussion_r246170274) [](ancestors = 246170274)